locate/word.go

```diff
@@ -47,7 +47,7 @@ func NextWordStart(textTree *text.Tree, pos uint64, targetCount uint64, stopAtEn
 
 		isWhitespace := gc.IsWhitespace()
 		hasNewline := gc.HasNewline()
-		isPunct := isPunct(gc)
+		isPunct := !withPunctuation && isPunct(gc)
```

Comment:
I think this isn't sufficient, because `prevWasPunct` might be true, in which case the condition on line 53 will be true.

Instead of modifying `isPunct`, consider modifying the conditions on lines 53 and 54 instead:
```
if (prevWasWhitespace && !isWhitespace) ||
(!withPunctuation && prevWasPunct && !isPunct && !isWhitespace) ||
(!withPunctuation && !prevWasPunct && isPunct) ||
(prevHasNewline && hasNewline)
```

Reply:
Done